### DISCUSSION TITLE
How to hide the window in the taskbar?

### DISCUSSION THREAD

**Discussion by Horbin-Magician** [slint-ui/slint#3266] [Q&A]
I want to write a pop-up search program. It's needless to be shown in taskbar. How can I hide it?
I try to use PopupWindow, but it can't  be the top level.
There is any method can help me hide me window in the taskbar or just change the flag of window?
> System: Windows11; Language: Rust.

**Comment by tronical** [comment]
At the moment this isn't possible out of the box :-(. One thing that you could try is for a given `slint::Window` to use the [`i-slint-backend-winit`](https://docs.rs/i-slint-backend-winit/latest/i_slint_backend_winit/) crate to obtain access to the `winit::window::Window` and then try to change the state this way.

**Reply by Horbin-Magician** [reply]
Thanks for your reply.  I try to use win32 to set the window's flag. The code is as follow:

```
search_win.window().with_winit_window(|winit_window: &winit::window::Window| {
    let raw_window_handle = winit_window.raw_window_handle();
    if let raw_window_handle::RawWindowHandle::Win32(win32_window_handle) = raw_window_handle {
        let hwnd = win32_window_handle.hwnd as isize;
        unsafe{
            WindowsAndMessaging::SetWindowLongPtrA(hwnd, WindowsAndMessaging::GWL_EXSTYLE, WindowsAndMessaging::WS_EX_TOOLWINDOW as isize);
            let result = WindowsAndMessaging::GetWindowLongPtrA(hwnd, WindowsAndMessaging::GWL_EXSTYLE);
        }
    }
});
```

But I failed to hide my window in the taskbar. Maybe I did something wrong, or the window flag is changed somewhere else.

I think this has little to do with Slint. The reply here is only to provide some references for others, and I also hope that people with solutions can help me.

**Reply by Vadoola** [reply]
If I recall I got something similar working on a program I am working on using a line similar to this

`i_slint_backend_winit::WinitWindowAccessor::with_winit_window(min_handle.window(), |win| win.set_visible(false));`

I was using it to "minimize" the window to a try icon.

**Comment by ogoffart** [comment]
I think hinhding the window is fairly simple, with  [`window().hide()`](https://docs.rs/slint/latest/slint/struct.Window.html#method.hide)

But the problem is to do that and keep the event loop running when there are no window.
You found the issue https://github.com/slint-ui/slint/issues/1499 already.
There are private API for this but we need to find a solution to make it work with public API.

**Reply by Horbin-Magician** [reply]
Thanks for your reply, and maybe some things are not clear enough. I just want my window don't show in the taskbar (may be a special case in Windows system). In Qt, I can easily realize it by:

``` c++
setWindowFlags(Qt::FramelessWindowHint | Qt::WindowStaysOnTopHint | Qt::Tool | Qt::Popup);
```

Of course, I also want to keep the event loop running when there are no window. 
I don't know how to use the private API (set_event_loop_quit_on_last_window_closed(false)). Hope to get more detailed guidance.

Thank you for providing such a great UI framework and for answering our questions so carefully.

**Reply by ogoffart** [reply]
Indeed i misunderstood.  Thanks for the clarification.

Yes, you will need to access winit directly as discussed in the other comment. (which is unfortunately still private API only)
